tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -434,20 +437,73 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	iptConfigurator.handleInboundIpv4Rules(ipv4RangesInclude)
 	iptConfigurator.handleInboundIpv6Rules(ipv6RangesExclude, ipv6RangesInclude)
 
-	// Execute iptables commands
-	for _, cmd := range iptConfigurator.iptables.BuildV4() {
+	iptConfigurator.executeCommands()
+}
+
+func (iptConfigurator *IptablesConfigurator) createRulesFile(f *os.File, contents string) error {
+	defer f.Close()
+	writer := bufio.NewWriter(f)
+	_, err := writer.WriteString(contents)
+	if err != nil {
+		return fmt.Errorf("Unable to write iptables-restore file: %v", err)
+	}
+	err = writer.Flush()
+	return err
+}
+
+func (iptConfigurator *IptablesConfigurator) executeIptableCommands(commands [][]string) {
+	for _, cmd := range commands {
 		if len(cmd) > 1 {
 			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
 		} else {
 			iptConfigurator.ext.RunOrFail(cmd[0])
 		}
 	}
+}
 
-	for _, cmd := range iptConfigurator.iptables.BuildV6() {
-		if len(cmd) > 1 {
-			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
-		} else {
-			iptConfigurator.ext.RunOrFail(cmd[0])
+func (iptConfigurator *IptablesConfigurator) executeIptableRestoreCommand(isIpv4 bool) error {
+	var data, filename, cmd string
+	if isIpv4 {
+		data = iptConfigurator.iptables.BuildV4Restore()
+		filename = fmt.Sprintf("iptables-rules-%d.txt", time.Now().UnixNano())
+		cmd = constants.IPTABLESRESTORE
+	} else {
+		data = iptConfigurator.iptables.BuildV6Restore()
+		filename = fmt.Sprintf("ip6tables-rules-%d.txt", time.Now().UnixNano())
+		cmd = constants.IP6TABLESRESTORE
+	}
+	rulesFile, err := ioutil.TempFile("", filename)
+	defer os.Remove(rulesFile.Name())
+	if err != nil {
+		return fmt.Errorf("Unable to create iptables-restore file: %v", err)
```

Comment:
Normally, error text isn't capitalized:
```suggestion
		return fmt.Errorf("unable to create iptables-restore file: %v", err)
```